Flow Aggregator panic due to divide-by-zero when EndTs is nil in aggregateRecords

### **Summary**

The Flow Aggregator can panic due to a divide-by-zero in aggregateRecords when processing flow records with a missing (nil) EndTs timestamp. This results in a crash of the flow-aggregator Pod.

### Bug Description

In aggregateRecords, throughput is calculated using the difference between incomingRecord.EndTs and existingRecord.EndTs.

However, if either timestamp is nil, the logic that computes flowEndSecondsDiff is skipped entirely. Despite this, the function proceeds to compute throughput using flowEndSecondsDiff, which remains 0, resulting in a division-by-zero panic.

## Impact

Severity: High (Denial of Service)

This bug allows a malformed or incomplete flow record to crash the Flow Aggregator, which is a centralized component responsible for collecting and exporting network flow data.

A faulty agent or malformed input could repeatedly trigger crashes, leading to loss of observability and disruption of flow analytics.

Even if current agents typically populate `EndTs`, this represents a lack of defensive validation and can be triggered by malformed input, partial flow export, or future regressions.
### Reproduction
Deploy Antrea with Flow Aggregator enabled
Ensure a flow record exists for a given flow key
Send an updated flow record with EndTs = nil (e.g., via malformed gRPC input or agent bug)
Observe flow-aggregator crash with:
panic: runtime error: integer divide by zero
### Expected Behavior
Gracefully handle missing timestamps
Skip throughput calculation or ignore invalid record
### Actual Behavior
Division by zero occurs
Flow Aggregator crashes
### Root Cause
if incomingRecord.EndTs != nil && existingRecord.EndTs != nil {
    flowEndSecondsDiff = incomingVal - prevFlowEndSeconds
}

// flowEndSecondsDiff == 0

throughput := totalCountDiff * 8 / uint64(flowEndSecondsDiff) // panic

The safety check is only applied inside the conditional block, but the division happens unconditionally.

### Why this is not already reported

This issue requires a specific combination of:

Optional protobuf fields (nil timestamps)
Control flow bypassing validation
Arithmetic depending on skipped logic

No existing issues reference this failure mode.

### Proposed Fix
if flowEndSecondsDiff > 0 {
    throughput = totalCountDiff * 8 / uint64(flowEndSecondsDiff)
    reverseThroughput = reverseTotalCountDiff * 8 / uint64(flowEndSecondsDiff)
}
### Test Plan
Add unit test with EndTs = nil
Verify no panic occurs
Ensure normal flows still compute throughput

--- a/pkg/flowaggregator/intermediate/aggregate.go
+++ b/pkg/flowaggregator/intermediate/aggregate.go
@@ -342,6 +342,11 @@ func (a *aggregationProcess) IsAggregatedRecordIPv4(record AggregationFlowRecord
 // addOrUpdateRecordInMap either adds the record to flowKeyMap or updates the record in
 // flowKeyMap by doing correlation or updating the stats.
 func (a *aggregationProcess) addOrUpdateRecordInMap(flowKey *FlowKey, record *flowpb.Flow, isIPv4 bool) {
+	if record.EndTs == nil || record.StartTs == nil {
+		klog.V(4).InfoS("Dropping flow record with nil timestamps", "record", record)
+		return
+	}
+
 	a.mutex.Lock()
 	defer a.mutex.Unlock()
 
@@ -595,8 +600,12 @@ func (a *aggregationProcess) aggregateRecords(incomingRecord, existingRecord *fl
 	// Update the throughput & reverseThroughput fields:
 	// throughput = (octetTotalCount - prevOctetTotalCount) / (flowEndSeconds - prevFlowEndSeconds)
 	// reverseThroughput = (reverseOctetTotalCount - prevReverseOctetTotalCount) / (flowEndSeconds - prevFlowEndSeconds)
-	throughput := totalCountDiff * 8 / uint64(flowEndSecondsDiff)
-	reverseThroughput := reverseTotalCountDiff * 8 / uint64(flowEndSecondsDiff)
+	var throughput, reverseThroughput uint64
+
+	if flowEndSecondsDiff > 0 {
+		throughput = totalCountDiff * 8 / uint64(flowEndSecondsDiff)
+		reverseThroughput = reverseTotalCountDiff * 8 / uint64(flowEndSecondsDiff)
+	}
 	if fillSrcStats {
 		existingRecord.Aggregation.ThroughputFromSource = throughput
 		existingRecord.Aggregation.ReverseThroughputFromSource = reverseThroughput

--- a/pkg/flowaggregator/intermediate/aggregate.go
+++ b/pkg/flowaggregator/intermediate/aggregate.go
@@ -601,7 +601,6 @@ func (a *aggregationProcess) aggregateRecords(incomingRecord, existingRecord *fl
 	// throughput = (octetTotalCount - prevOctetTotalCount) / (flowEndSeconds - prevFlowEndSeconds)
 	// reverseThroughput = (reverseOctetTotalCount - prevReverseOctetTotalCount) / (flowEndSeconds - prevFlowEndSeconds)
 	var throughput, reverseThroughput uint64
-
 	if flowEndSecondsDiff > 0 {
 		throughput = totalCountDiff * 8 / uint64(flowEndSecondsDiff)
 		reverseThroughput = reverseTotalCountDiff * 8 / uint64(flowEndSecondsDiff)

